src/main/scala/shine/OpenCL/compilation/HostManagedBuffers.scala

```diff
@@ -0,0 +1,271 @@
+package shine.OpenCL.compilation
+
+import shine.DPIA._
+import shine.DPIA.Phrases._
+import shine.DPIA.Semantics.OperationalSemantics
+import shine.DPIA.Types._
+import shine.DPIA.primitives.{imperative => dpia}
+import shine.OpenCL._
+import shine.OpenCL.primitives.{imperative => ocl}
+
+import scala.annotation.tailrec
+import scala.collection.{immutable, mutable}
+
+object HostManagedBuffers {
+  def populate(params: immutable.Seq[Identifier[ExpType]], outParam: Identifier[AccType])
```

Comment:
Can you give a brief comment explaining the behaviour of this function. What is populated with what?

Reply:
What about: "Given a host program which only operates on plain arrays, this pass populates host managed buffers where needed, as well as host execution sections where plain arrays can still be used."?

Reply:
Thanks for adding the comment.